engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/error/CatchPanicNode.java

```diff
@@ -59,7 +59,7 @@ Object doExecute(
       @Cached BranchProfile otherExceptionBranchProfile,
       @CachedLibrary(limit = "3") InteropLibrary interop) {
     try {
-      return thunkExecutorNode.executeThunk(frame, action, state, BaseNode.TailStatus.TAIL_DIRECT);
+      return thunkExecutorNode.executeThunk(frame, action, state, BaseNode.TailStatus.NOT_TAIL);
```

Comment:
`Panic.catch` cannot be in tail call position. I believe it has to be `NOT_TAIL`. All the tests in `Error_Spec` are fixed. Let's see what other _collateral damage_ this may created. Btw. I have a feeling @hubertp was fighting with some similar issue a year back or so...

Reply:
What a good observation! At the end, it seems that we will not have to dig in the IR phases. I have just added some comments and made sure that even the `handler` callback is not in a tail call position, just to be sure - in commit 85399979ebfdc0987c3c69a6d5bf6f4747b089f8.

Reply:
On contrary to the `action` invocation, handler can probably be in tail position. There is nothing in the `CatchPanicNode` that needs to run after handler...

Reply:
Good note. Add a comment about that in d86c7b5467627cfc15edaca732e90e766bcb4fec . `handler` is invoked with `TAIL_DIRECT`.

Reply:
> Btw. I have a feeling @hubertp was fighting with some similar issue a year back or so...

It's been a while so I can't remember but maybe you are referring to https://github.com/enso-org/enso/pull/7116